### PR TITLE
Add unit and sanity test workflows to GHA

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY

This PR adds unit and sanity test workflows to Github Actions.

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by alinabuzachis** [comment]
@GomathiselviS merging this https://github.com/ansible-collections/amazon.cloud/pull/54 should solve some of the issues.

**Comment by gravesm** [comment]
If the 3.8 tests are passing, I don't see any harm in it. If we get to the point where they start failing, we can add an exception for this repo.

File: plugins/module_utils/utils.py [hunk 1]
```diff
@@ -23,9 +23,10 @@
     recursive_diff,
 )
 
+# pylint: disable=unused-import
 from ansible.module_utils.common.dict_transformations import (
     snake_dict_to_camel_dict,
-)  # pylint: disable=unused-import
```
Comment (alinabuzachis) [inline plugins/module_utils/utils.py:28]: It should work this way with two spaces before adding this `# pylint: disable=unused-import`

File: test-requirements.txt [hunk 2]
```diff
@@ -6,3 +6,4 @@ placebo
 mock
 pytest
 pytest-xdist
+pytest-ansible-units ; python_version > '3.8'
```
Comment (alinabuzachis) [inline test-requirements.txt:9]: Why would we need to add python_version dependency here?

**Review by alinabuzachis — APPROVED** [review]

File: test-requirements.txt (hunk 2 above)
Comment (GomathiselviS) [reply test-requirements.txt:9]: I shall remove it. It was a copy paste mistake.

File: plugins/module_utils/utils.py (hunk 1 above)
Comment (GomathiselviS) [reply plugins/module_utils/utils.py:28]: It does not for some Python versions. Hence changed the format.

Comment (GomathiselviS) [reply plugins/module_utils/utils.py:28]: The failures for some of the py versions are seen here - https://github.com/ansible-collections/amazon.cloud/actions/runs/4801456605

**Pull request merged**